src/librustc/ty/error.rs

```diff
@@ -222,7 +222,22 @@ impl<'a, 'gcx, 'lcx, 'tcx> ty::TyS<'tcx> {
             ty::TyArray(_, n) => format!("array of {} elements", n),
             ty::TySlice(_) => "slice".to_string(),
             ty::TyRawPtr(_) => "*-ptr".to_string(),
-            ty::TyRef(_, _) => "&-ptr".to_string(),
+            ty::TyRef(region, tymut) =>
+                match tymut {
+                    ty::TypeAndMut{ref ty, ref mutbl} => {
```

Comment:
This whole `match` is unnecessary, we'd rather try to pretend `TypeAndMut` doesn't exist - you can put this pattern in the `TyRef` pattern if you want; the by-ref matching is also unnecessary.